### PR TITLE
Added support for php8

### PR DESCRIPTION
**Pull request by maxmilash**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,14 +16,14 @@
         }
     ],
     "require": {
-        "php": "^5.6 || ^7.0",
+        "php": "^5.6 || ^7.0 || ^8.0",
```
Comment (garak) [inline composer.json:19]: Let's remove support for php 5

File: composer.json [hunk 2]
```diff
@@ -16,14 +16,14 @@
...
         "liuggio/statsd-php-client": "~1.0.14",
         "symfony/framework-bundle": "^3.0 || ^4.0 || ^5.0"
```
Comment (garak) [inline composer.json:21]: Let's remove support for Symfony 3 as well

**Review by garak — CHANGES_REQUESTED** [review]

File: composer.json [hunk 3]
```diff
@@ -16,15 +16,15 @@
...
         "liuggio/statsd-php-client": "~1.0.14",
-        "symfony/framework-bundle": "^3.0 || ^4.0 || ^5.0"
+        "symfony/framework-bundle": "^4.0 || ^5.0"
     },
     "require-dev": {
         "doctrine/dbal": ">=2.2.0",
         "monolog/monolog": ">=1.2.0",
-        "phpunit/phpunit": "^5.7 || ^6.0 || ^7.0",
-        "symfony/yaml": "^3.0 || ^4.0 || ^5.0"
+        "phpunit/phpunit": "^5.7 || ^6.0 || ^7.0 || ^9.5",
```
Comment (garak) [inline composer.json:26]: We don't need any phpunit below ^6.5

**Review by garak — CHANGES_REQUESTED** [review]

**Review by garak — APPROVED** [review]

**Comment by maxmilash** [comment]
@garak When can expect a release approximately for current PR?

**Comment by Phobetor** [comment]
@garak Out of curiosity: why did you request to drop support for the old versions?

I know, they are not maintained anymore. But they do not pose any incompatibilities as far as I can see.
So why build any walls that could prevent user from using the most current version of this bundle?

**Comment by garak** [comment]
> @garak Out of curiosity: why did you request to drop support for the old versions?
> 
> I know, they are not maintained anymore. But they do not pose any incompatibilities as far as I can see. So why build any walls that could prevent user from using the most current version of this bundle?

It's just to push forward the ecosystem. It's better to keep only the actively maintained versions.
Anyone using an old version can just keep using old versions of this bundle as well.

**Comment by Phobetor** [comment]
> It's just to push forward the ecosystem. It's better to keep only the actively maintained versions.

In general I support this idea. There are some specifics with this bundle and its releases that make it a special case.

> Anyone using an old version can just keep using old versions of this bundle as well.

That is exactly the problem because it doesn't work.

Since 2018 the latest release version has been [1.7.0](https://github.com/liuggio/StatsDClientBundle/commit/f1d11b9adc7257c06dc0c679267daf8661788a96).
Since then everyone using this bundle with symfony `3.*` needs to use `dev-master` instead of a release version.

Now there is a new release version `1.8.0` which requires symfony `>=4`. So again, no luck for symfony `3.*`.

The obvious solution would be: staying with `dev-master` and pinning the commit before the change that excluded symfony `3.x`. But this is not working because [composer is always looking at the newest commit of the used branch instead of the pinned commit when checking dependencies](https://getcomposer.org/doc/articles/troubleshooting.md#i-have-locked-a-dependency-to-a-specific-commit-but-get-unexpected-results-).

Long story short: there is no way to use this package with symfony `3.*` anymore. I need to fork it, allow symfony `3.*` and then use that fork.

Just my two cents about keeping compatibility paths:
In general in my opinion it would have been better practice to first deploy the functional changes, release that state as a new version and then in a second step do the upgrade-path-cutting operations and release another version. This could have been connected with a major version bump to keep a way open for later fixes on the old version.

**Comment by garak** [comment]
You should _never_ use `dev-master`, unless you're forced to.
For this bundle, you can use `^1.8@dev`

**Comment by Phobetor** [comment]
> For this bundle, you can use ^1.8@dev

As I wrote, this is not possible with symfony 3.

Here you can see the composer output for having symfony 3 and requiring liuggio/statsd-client-bundle >=1.7.0 (so basically all reasonable versions):
```
./composer.json has been updated
Running composer update liuggio/statsd-client-bundle
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - liuggio/statsd-client-bundle[dev-master, v1.8.0] require symfony/framework-bundle ^4.0 || ^5.0 -> found symfony/framework-bundle[v4.0.0-BETA1, ..., 4.4.x-dev, v5.0.0-BETA1, ..., 5.4.x-dev] but these were not loaded, likely because it conflicts with another require.
    - liuggio/statsd-client-bundle v1.7.0 requires symfony/framework-bundle ^2.3 -> found symfony/framework-bundle[v2.3.0, ..., 2.8.x-dev] but these were not loaded, likely because it conflicts with another require.
    - liuggio/statsd-client-bundle 1.8.x-dev is an alias of liuggio/statsd-client-bundle dev-master and thus requires it to be installed too.
    - Root composer.json requires liuggio/statsd-client-bundle >=1.7.0 -> satisfiable by liuggio/statsd-client-bundle[v1.7.0, v1.8.0, 1.8.x-dev (alias of dev-master)].

Installation failed
```

`satisfiable by liuggio/statsd-client-bundle[v1.7.0, v1.8.0, 1.8.x-dev (alias of dev-master)]`
=> checked 1.7, 1.8 and even dev-master.

All of them have requirements conflicting with symfony 3.

Conclusion:
> Long story short: there is no way to use this package with symfony 3.* anymore. I need to fork it, allow symfony 3.* and then use that fork.

I did this now, and that works. I would still prefer a more backwards compatible release flow as I described above.
